waspc/data/Generator/templates/react-app/src/auth/forms/internal/email/useEmail.ts

```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react'
+import { signup } from '../../../email/actions/signup'
+import { login } from '../../../email/actions/login'
+
+export function useEmail({
+  onError,
+  showEmailVerificationPending,
+  isEmailVerificationRequired,
+  onLoginSuccess,
+  isLogin,
+}: {
+  onError: (error: Error) => void
+  showEmailVerificationPending: () => void
+  onLoginSuccess: () => void
+  isLogin: boolean
+  isEmailVerificationRequired: boolean
+}) {
+  const [emailFieldVal, setEmailFieldVal] = useState('')
+  const [passwordFieldVal, setPasswordFieldVal] = useState('')
+
+  async function handleSubmit() {
+    try {
+      if (isLogin) {
+        await login({ email: emailFieldVal, password: passwordFieldVal })
+        onLoginSuccess()
+      } else {
+        await signup({ email: emailFieldVal, password: passwordFieldVal })
+        if (isEmailVerificationRequired) {
```

Comment:
I wanted to reduce the number of files that need to be templates. 

Here, we have a function called `useEmail` which is exported from `internal/email/useEmail.ts` and since it receives a param `isEmailVerificationRequired` there is not need for it to be a Mustache template. 

Every time we have a Mustache template, VS Code complains and doesn't color the Typescript nicely, so, the less we use then, the better :)

Reply:
That makes sense!